src/Http/Controllers/DuplicateController.php

```diff
@@ -14,40 +21,70 @@ public function duplicate(Request $request)
     {
         // Replicate the model
         $model = $request->model::where('id', $request->id)->first();
-
+        
         if (!$model) {
             return [
                 'status' => 404,
                 'message' => 'No model found.',
                 'destination' => config('nova.url') . config('nova.path') . '/resources/' . $request->resource . '/'
             ];
         }
-
+        
         $newModel = $model->replicate();
         $newModel->push();
-
+        
         if (isset($request->relations) && !empty($request->relations)) {
             // load the relations
             $model->load($request->relations);
-
+            
             foreach ($model->getRelations() as $relation => $items) {
-                // works for hasMany
-                foreach ($items as $item) {
-                    // clean up our models, remove the id and remove the appends
-                    unset($item->id);
-                    $item->setAppends([]);
-
-                    // create a relation on the new model with the data.
-                    $newModel->{$relation}()->create($item->toArray());
+                $relation_data = $model->{$relation}();
+                switch(get_class($relation_data)){
+                    case BelongsToMany::class:
+                    case MorphToMany::class:
+                        foreach ($items as $item) {
```

Comment:
Probably want to extract these to individual functions to clean up this, when all classes are added it could become hard to read/manage.

Reply:
As you wish. I think you can merge this changes to this currently version of your repository.

Reply:
Seems good, I'll give it a test and double-check